estate/models/estate_property.py

```diff
@@ -0,0 +1,102 @@
+from datetime import date
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+    _order = "id desc"
+
+    name = fields.Char(string="Title", required=True)
+    property_type = fields.Selection(
+        string='Property Type',
+        selection=[('house', 'House'), ('apartment', 'Apartment')])
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, readonly=True, default=lambda self: date.today() + relativedelta(months=3))
+    expected_price = fields.Float("Expected Price", required=True)
+    selling_price = fields.Float("Selling Price", readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer(string='Living Area (m2)')
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer(string='Garden Area (m2)')
+    garden_orientation = fields.Selection(
+        string='Garden Orientation',
+        selection=[('north', 'North'), ('south', 'South'),
+        ('east', 'East'), ('west', 'West')])
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+    copy=False,
+    readonly=True,
+    default='new',
+    string='States',
+    selection=[
+        ('new', 'New'),
+        ('offer_received', 'Offer Received'),
+        ('offer_accepted', 'Offer Accepted'),
+        ('sold', 'Sold'),
+        ('cancelled', 'Cancelled')]
+    )
+
+    def action_set_sold(self):
+        if self.state != "cancelled":
+            self.state = "sold"
+        else:
+            raise UserError("A cancelled property can not be sold")
+        return True
```

Comment:
```suggestion
    def action_set_sold(self):
        if self.state == "cancelled":
            raise UserError("A cancelled property can not be sold")
        self.state = "sold"
        return True
```
More concise and extendable

Reply:
Also what would happen here if self contains more than 1 property ?